azurerm_redis_cache/vars.tf

```diff
@@ -0,0 +1,97 @@
+variable "global_prefix" {
+  type = string
+  description = "Prefix used to define the resource name."
+}
+
+variable "environment" {
+  type          = string
+  description   = "The name of the environment"
+}
+
+variable "region" {
+  type = string
+  description = "The location of the resource group."
+}
+
+variable "name" {
+  type        = string
+  description = "The name of the Redis instance."
+}
+
+variable "resource_group_name" {
+  type = string
+}
+
+variable "capacity" {
+  type          = number
+  description   = "The size of the Redis cache to deploy"
+  default       = 1
+}
+
+variable "shard_count" {
+    type        = number
+    description = "The number of Shards to create on the Redis Cluster."
+    default     = null
+}
+
+variable "enable_non_ssl_port" {
+    type        = bool
+    description = "Enable the non-SSL port (6379) - disabled by default."
+    default     = false
+}
+
+variable "minimum_tls_version" {
+    type        = string
+    description = "The minimum TLS version."
+    default     = "1.0"
+}
+
+variable "subnet_id" {
+    type        = string 
+    description = "The Subnet within which the Redis Cache should be deployed."
+    default     = "" 
+}
+
+variable "private_static_ip_address" {
+    type        = string
+    description = "The Static IP Address to assign to the Redis Cache when hosted inside the Virtual Network"
+    default     = ""
+}
+
+variable "family" {
+    type        = string
+    description = "The SKU family/pricing group to use"
+}
+
+variable sku_name {
+    type        = string
+    description = "The SKU of Redis to use"
+}
+
+variable "rdb_backup_enabled" {
```

Comment:
If rdb_backup_enabled = true than rdb_backup_frequency, rdb_backup_max_snapshot_count, rdb_storage_connection_string are required.
I haven't checked if is possible to specify a default value for an object type variable.
Please try to use a single variable of type object(rdb_backup_frequency, rdb_backup_max_snapshot_count, rdb_storage_connection_string) with default of null and than:
- if == null set rdb_backup_enabled to false
- if != null set rdb_backup_enabled to true and also the other values

Reply:
I am not sure about it since  - if I am not wrong -  **redis_configuration** in main.tf is not an object. It's a block .... I'll play with it a bit ....